src/buildblock/SSRB.cxx

```diff
@@ -76,17 +77,23 @@ SSRB(const ProjDataInfo& in_proj_data_info,
       num_tang_poss_to_trim)
     error("SSRB: too large number of tangential positions to trim (%d)\n",
 	  num_tang_poss_to_trim);
+
   const ProjDataInfoCylindrical * const in_proj_data_info_sptr =
     dynamic_cast<ProjDataInfoCylindrical const * >
     (&in_proj_data_info);
-  if (in_proj_data_info_sptr== NULL)
-    {
-      error("SSRB works only on segments with proj_data_info of "
-	    "type ProjDataInfoCylindrical\n");
-    }
-  ProjDataInfoCylindrical * out_proj_data_info_sptr =
+    ProjDataInfoCylindrical * out_proj_data_info_sptr =
     dynamic_cast<ProjDataInfoCylindrical * >
     (in_proj_data_info_sptr->clone());
+  if (in_proj_data_info_sptr== NULL && out_proj_data_info_sptr==NULL)
```

Comment:
```suggestion
  if (!in_proj_data_info_sptr )
```
is more C++11 like. you'll get warnings otherwise for some compilers anyway. Also no need to test for `out` as it's a `clone`